### ISSUE TITLE
Consider forking the web UI to make an API server out of it

### ISSUE THREAD

**Issue by philpax** [philpax/stable-diffusion-a1111-webui-client#3]
The A1111 web UI has a lot of features I don't care about (namely the web UI), and the API is not the best, especially for multi-tenant scenarios. The user experience might be improved if I forked it, excised all of the web UI stuff, and built a solid API on top of it, hopefully in such a way that keeping the fork inline with upstream is doable.

Features I'd want:
- Model switching in queue: https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/5847
- Generation jobs getting their own ID that specific progress can be queried for: https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/3786
- Consistent API design
- No leaky abstractions
- Expose all configuration (including embeddings) through the API
- Remove any features designed for users (artists, prompt styles)
- Queue manipulation functions, including viewing and deleting jobs

**Comment by dsully** [comment]
Don't forget - rewritten in Rust, perhaps using PyO3 to interface with torch and diffusers. :)

**Comment by philpax** [comment]
I'd much rather use https://github.com/LaurentMazare/diffusers-rs in that case ;)

**Comment by dsully** [comment]
Fair enough!

**Comment by dcvz** [comment]
Have you done anything towards the queue jobs via API @philpax ?

**Comment by philpax** [comment]
No, I'm afraid not. I figured I'd wait and see if someone else built a decent API server.

**Comment by philpax** [comment]
No, I'm afraid not. I figured I'd wait and see if someone else built a decent API server.
